packages/app/pages/[modelName]/index.tsx

```diff
@@ -115,7 +116,7 @@ const ModelPage = ({ user, model, allModels, documents }: ModelPageProps) => {
                 {documents === null ? (
                     <p className="text-center py-4 text-danger">
                         mEditor had an error getting documents for{' '}
-                        {model.name || 'this model'}. Please verify that your{' '}
+                        {model?.name || 'this model'}. Please verify that your{' '}
```

Comment:
Can't assume the model actually loaded, without the `?.` this throws a JS error which breaks the page

Reply:
For what it's worth, I can't get this to break since you added the handling of `modelError` as a page-level 404 in `getServerSideProps`. If that's the case for you too, I'd be in favor of removing it since our typing doesn't list model as an optional property and (or null). I think optional property accessors are kind of a concession: "We couldn't do any better with data surety or assertions, so we're just not sure."

Reply:
Actually this is revealing an issue that we've encountered a few times, and that's assuming something is defined based off types. The problem is, we have to also guarantee `model` is defined at **run-time**, which we can't here.

So I wouldn't remove this until we can assert that. Without the `?.`, if the database goes down, the UI will crash with a JS error. That means we're not handling certain error states (coding to the happy path). And database downtime is only one potential error.

A way to handle this would be to refactor this component to have some sort of a conditional before this section that returns early if the model is not defined. That way we can guarantee it exists.

Side note, I would disagree with `?.`'s being a concession, to be honest. Yes, they can be overused, but there are also legitimate uses for them. This is one of those: we can avoid page breaking JS errors by using them in areas where we can't guarantee run-time existence of an object.